### PR TITLE
[G2M] add Input base on delete prop

### PR DESCRIPTION
**Pull request by kc-leung**
- Added on click delete prop for input/textfield component

### CODE REVIEW

File: src/base-components/input-base.js [hunk 1]
```diff
@@ -16,6 +16,7 @@ const InputBase = forwardRef(({
   placeholder,
   onClick,
   onChange,
+  onDelete,
```
Comment (woozyking) [inline src/base-components/input-base.js:19]: is there a way to fully relay "native" `input` event interfaces?

**Review by woozyking — APPROVED** [review]

File: src/base-components/input-base.js (hunk 1 above)
Comment (woozyking) [reply src/base-components/input-base.js:19]: this is just out of curiosity. what we have is fine

Comment (kc-leung) [reply src/base-components/input-base.js:19]: not sure will have to test it out

File: src/components/dropdown-auto-complete.js [hunk 2]
```diff
@@ -105,7 +105,7 @@ const DropdownAutoComplete = ({
     setComponentIsActive((state) => !state)
     setOpen(!open)
   }
-
+
```
Comment (woozyking) [inline src/components/dropdown-auto-complete.js:108]: make sure to configure your editor to auto-remove empty trailing spaces

**Review by woozyking — APPROVED** [review]

**Review by hyx131 — APPROVED** [review]

**Pull request merged**